src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchIntegrationTests.java

```diff
@@ -1499,6 +1514,33 @@ public void shouldDeleteIndexForGivenEntity() {
 		assertThat(indexOperations.exists()).isFalse();
 	}
 
+	@Test
+	public void shouldDoPartialUpdateBySuppliedEntityForExistingDocument() {
+
+		// given
+		String documentId = nextIdAsString();
+		String messageBeforeUpdate = "some test message";
+		String messageAfterUpdate = "test message";
+
+		SampleEntity sampleEntity = SampleEntity.builder().id(documentId).message(messageBeforeUpdate)
+				.version(System.currentTimeMillis()).build();
+
+		IndexQuery indexQuery = getIndexQuery(sampleEntity);
+
+		operations.index(indexQuery, IndexCoordinates.of(indexNameProvider.indexName()));
+
+		// modify the entity
+		sampleEntity.setMessage(messageAfterUpdate);
+
+		// when
+		operations.update(sampleEntity);
+
+		// then
+		SampleEntity indexedEntity = operations.get(documentId, SampleEntity.class,
+				IndexCoordinates.of(indexNameProvider.indexName()));
+		assertThat(indexedEntity.getMessage()).isEqualTo(messageAfterUpdate);
```

Comment:
```suggestion
		// given
		String documentId = nextIdAsString();
		String messageBeforeUpdate = "some test message";
		String messageAfterUpdate = "test message";

		SampleEntity sampleEntity = SampleEntity.builder().id(documentId).message(messageBeforeUpdate).type("some type")
				.version(System.currentTimeMillis()).build();
		operations.save(sampleEntity);

		// modify the entity
		sampleEntity.setMessage(messageAfterUpdate);
		sampleEntity.setType(null);

		// when
		operations.update(sampleEntity);

		// then
		SampleEntity indexedEntity = operations.get(documentId, SampleEntity.class);
		assertThat(indexedEntity.getType()).isEqualTo("some type");
		assertThat(indexedEntity.getMessage()).isEqualTo(messageAfterUpdate);
```

Just modifying one property is not testing the partial update, you'd get the same result when doing a normal `save`. You need do actually set some property to `null` to make the update partial and check, thta this null property was not updated.